### PR TITLE
Add integration tests for the upgrade process

### PR DESCRIPTION
**Pull request by zymap**
**Motivation**

The streamnative/charts can upgrade from the apache/pulsar-helm-charts 
to support more components provided in the streamnative/charts.
Add an integration test for the default deploy upgrade check.

### CODE REVIEW

File: .ci/clusters/values-apache-default.yaml [hunk 1]
```diff
@@ -0,0 +1,1019 @@
+#
```
Comment (sijie) [inline .ci/clusters/values-apache-default.yaml:1]: Can you avoid adding the entire default values file? You should just add the changes here. Please refer to other value files under the `.ci/clusters` directory.

File: .ci/clusters/values-apache-upgrade.yaml [hunk 2]
```diff
@@ -0,0 +1,1023 @@
+#
```
Comment (sijie) [inline .ci/clusters/values-apache-upgrade.yaml:1]: What do we need a separate `upgrade values file? Can we use the `values-apache-default.yaml` to upgrade?

**Review by sijie — COMMENTED** [review]
@zymap Can you

**Comment by zymap** [comment]
@sijie PTAL. Thanks

**Comment by zymap** [comment]
Merge this PR first, If has any problem, I will fix it in another PR

**Pull request merged**